api/v1alpha1/server_types.go

```diff
@@ -290,16 +290,44 @@ type NetworkInterface struct {
 	// +required
 	Name string `json:"name"`
 
-	// IP is the IP address assigned to the network interface.
-	// The type is specified as string and is schemaless.
-	// +kubebuilder:validation:Type=string
-	// +kubebuilder:validation:Schemaless
-	// +required
-	IP IP `json:"ip"`
+	// IPs is a list of IP addresses (both IPv4 and IPv6) assigned to the network interface.
+	// +optional
+	IPs []IP `json:"ips,omitempty"`
```

Comment:
This will be a breaking change. How about keeping the `ip` field and let the controller `nil`ify it for now and store the actual IP in the `ips` slice? We should also deprecate this `ip` field and remove it in a later version. @Nuckal777 @defo89 are you guys currently using rely on this field?

Reply:
we decided this is fine.

Reply:
As discussed, agree with @afritzler's suggestion. Lets keep the ip field, and set it to nil for now and deprecate it later, and we will remove the dependency from the boot operator.